Maven plugin

gunnarmorling opened the pull request:
Hey @tombentley, here's the first cut for the Maven plug-in. It's WIP, I'll create a PR which uses this over in kproxy in a bit, along with some questions for you.

tombentley commented on .gitignore:
```diff
@@ -1,4 +1,10 @@
+# IntelliJ
 .idea/
 target/
```
target is mvn not idea, so the comment doesn't apply to this one.

tombentley commented on src/main/java/com/github/tombentley/krpccodegen/main/KrpcGenerator.java:
```diff
@@ -208,9 +219,7 @@ private Configuration buildFmConfiguration() throws IOException {
-        // Specify the source where the template files come from. Here I set a
-        // plain directory for it, but non-file-system sources are possible too:
-        cfg.setDirectoryForTemplateLoading(templateDir);
+        cfg.setClassLoaderForTemplateLoading(templateClassLoader, "/");
```
Do we _really_ want to load templates via a class loader? Doesn't it imply that they have to be in a jar, rather than just files on the filesystem? I would have thought that it would be better to move the `Filter.ftl` to kproxy and have the maven plugin just look in `src/main/krpc-templates` in the kproxy repo.

gunnarmorling replied:
Very good question. In fact my thinking was that those templates are rather "static" and thus should be shipped via the plug-in (in which case we need to look them up via the plug-ins class loader). But you may be right and they are usage specific, in which case moving them over to kproxy makes absolute sense.

gunnarmorling commented:
Hey @tombentley, I think this is good to go. Reworked the initial PR to load templates from a directory again and added some more commits for some clean-up. Once this PR and its counterpart over in kproxy have been merged, I'd like to move the plug-in over to the kproxy repo, making the same a multi-module build.

tombentley commented on src/main/java/com/github/tombentley/krpccodegen/main/KrpcGenerator.java:
```diff
@@ -44,21 +45,68 @@
+            return this;
+        }
+
+        public Builder withSchemaDir(File schemaDir) {
```
I wonder whether 'schema' is the right term to use. I used it in `KrpcGenerator`, but the code I copied and modified from Kafka uses `MessageSpec` for the same concept, and in hindsight I realise that it's confusing having two names for the same thing. On the one hand Schema is shorter, obviously. But OTOH when copying `MessageSpec` and those classes from Kafka I tried to minimise the changes I made, so I don't think renaming `MessageSpec` → `Schema` would be the right move.

gunnarmorling replied:
Everything is fine for me. So you'd prefer `MessageSpec`, i.e. `withMessageSpecDir(File messageSpecDir) { ... }`? If so, no problem, I'll quickly update this.

tombentley replied:
Yeah, I think that would be best. Thanks!

gunnarmorling commented:
@tombentley, pushed a comment for the renaming, this should be good to go now. I'll update the corresponding PR over in the main repo, and once this one is merged, we can continue with everything over there.
